hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -262,6 +268,12 @@ public Response put(
         body = new SignedChunksInputStream(body);
       }
       long putLength = 0;
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      IOUtils.copyLarge(body, baos);
+      body = new ByteArrayInputStream(baos.toByteArray());
```

Comment:
If the file is very large, this byte array will be very large, it is recommended to change to a wrapped stream in which the md5 is calculated using streaming.

Reply:
+1

Reply:
With the use of DigestInputStream, this problem seems to be solved